core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -110,17 +110,28 @@ oppia.directive('searchBar', [function() {
         };
 
         var _searchBarFullyLoaded = false;
+        var url;
 
         var _hasChangedSearchQuery = Boolean(urlService.getUrlParams().q);
-        var _onSearchQueryChangeExec = function() {
-          searchService.executeSearchQuery(
+        var _onSearchQueryChangeExec = function(reloadPage) {
+          $scope.searchIsLoading = true;
+          url = searchService.executeSearchQuery(
```

Comment:
hmm, the reason I did this was that `searchService.executeSearchQuery` already had produced a link when it was called for when it sends a query to `/galleryhandler/`. I think a more elegant solution may be to extract those URL construction details into a helper function in `searchService`?